bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/bundlefile/CloseableBundleFile.java

```diff
@@ -73,17 +73,20 @@ public CloseableBundleFile(File basefile, BundleInfo.Generation generation, MRUB
 	 * @return true if the bundle file is open and locked
 	 */
 	protected boolean lockOpen() {
+		openLock.lock();
 		try {
-			open(true);
+			internalOpen();
 			return true;
-		} catch (IOException e) {
+		} catch (Throwable e) {
```

Comment:
You are swallowing all exceptions here. Before, you only swallowed IOException and all others flowed. I think you need rethrow non-IOExceptions here (after unlocking) to preventing Errors from being lost.

Reply:
Fair enough, I can throw the other exceptions.  I figured I was logging the other exceptions as errors which seems appropriate also.  But we can keep the old behavior.